app/Models/Game.php

```diff
@@ -292,17 +292,10 @@ public function registerMediaCollections(): void
 
     public function toSearchableArray(): array
     {
-        return $this->only([
-            'ID',
-            'Title',
-        ]);
-    }
-
-    public function shouldBeSearchable(): bool
-    {
-        // TODO return $this->isPublished();
-        // TODO return true;
-        return false;
+        return [
+            'id' => (int) $this->ID,
```

Comment:
Is this given precedence because it's first? If I search for "2025", I get like 10 matches where 2025 is part of the ID before I get any of the 2025 events.

Reply:
It is not given precedence because it's first. Anything returned by `toSearchableArray()` is a searchable value. Docs [here](https://laravel.com/docs/11.x/scout#configuring-searchable-data).

Ranking rules for a model are set up in the _config/scout.php_ file. [Docs reference](https://laravel.com/docs/12.x/scout#configuring-filterable-data-for-meilisearch). [Meilisearch settings reference](https://www.meilisearch.com/docs/reference/api/settings).

I believe `searchableAttributes` is the setting we're after, based on this text from the Meilisearch settings docs:
> Fields in which to search for matching query words sorted by order of importance

I've added it in latest with `title` prioritized over `id`.

Reply:
Searching on "2000" is still preferring game IDs:
![image](https://github.com/user-attachments/assets/6dc7bcac-6c1a-453f-9ed1-30dea87e27e3)
![image](https://github.com/user-attachments/assets/237c297e-0055-4a4b-9384-905cdbfb7818)

Reply:
https://github.com/RetroAchievements/RAWeb/pull/3491#discussion_r2083270336

After running the same command referenced in this other comment, I now see:
![Screenshot 2025-05-10 at 3 00 01 PM](https://github.com/user-attachments/assets/3f28fc00-7be6-4bc6-9137-9472c6fbaed2)

Reply:
This appears to be working, but I'm still having issues with users. I don't think the sync command actually did anything.